### ISSUE TITLE
bug: executing UpdateAssumeRolePolicy results in MalformedPolicyDocument

### ISSUE THREAD

**Issue by rodjul** [localstack/localstack#6644]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When running the latest image published to docker hub, the service fails to update the policy of role.

Task: I'm testing some creation of roles and updating their trust relationship policies.

Executing the following commands:
```sh
awslocal iam get-role --role-name teste123
awslocal iam create-role --role-name teste123 --assume-role-policy-document file://Test-Role-Trust-Policy.json
awslocal iam update-assume-role-policy --role-name teste123 --policy-document file://Test-Role-Trust-Policy.json
```
the command `update-assume-role-policy` fails saying the policy used contains invalid syntax.

Message of error: `An error occurred (MalformedPolicyDocument) when calling the UpdateAssumeRolePolicy operation: Has prohibited field Resource.`

Logs:
```
2022-08-10T14:51:12.537  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS iam.GetRole => 404 (NoSuchEntity); GetRoleRequest({'RoleName': 'teste123'}, headers={'Host': 'localhost:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/1.22.100 Python/3.8.10 Linux/5.10.102.1-microsoft-standard-WSL2 botocore/1.25.2', 'X-Amz-Date': '20220810T145112Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=a/20220810/us-east-2/iam/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=22a73d408201fd4110357f0567878b630bce7866a5587ef38ccf3ce06ea975ee', 'amz-sdk-invocation-id': '5ab92200-fe4c-4366-9060-390d2aae0c5c', 'amz-sdk-request': 'attempt=1', 'Content-Length': '51', 'x-localstack-tgt-api': 'iam'}); NoSuchEntity(Role teste123 not found, headers={'Content-Type': 'application/xml', 'server': 'amazon.com', 'X-Amzn-ErrorType': 'NoSuchEntity', 'status': '404', 'Content-Length': '307', 'Connection': 'close', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 'Access-Control-Allow-Headers': 'authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request', 'Access-Control-Expose-Headers': 'etag,x-amz-version-id'})

2022-08-10T14:51:18.644  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS iam.CreateRole => 200; CreateRoleRequest({'RoleName': 'teste123', 'AssumeRolePolicyDocument': '{\n    "Version": "2012-10-17",\n    "Statement": [\n        {\n            "Effect": "Allow",\n            "Principal": {\n                "Service": [\n                    "ec2.amazonaws.com"\n                ]\n            },\n            "Action": [\n                "sts:AssumeRole"\n            ]\n        }\n    ]\n}'}, headers={'Host': 'localhost:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/1.22.100 Python/3.8.10 Linux/5.10.102.1-microsoft-standard-WSL2 botocore/1.25.2', 'X-Amz-Date': '20220810T145118Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=a/20220810/us-east-2/iam/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=f6ecb6be101bf7b6269f305804a134e6d300f97f4752a54e4f9af8902c4e4a87', 'amz-sdk-invocation-id': 'b8890026-c1ed-4fe3-b3d4-26e21d5abf83', 'amz-sdk-request': 'attempt=1', 'Content-Length': '502', 'x-localstack-tgt-api': 'iam'}); CreateRoleResponse({'Role': {'Path': '/', 'RoleName': 'teste123', 'RoleId': 'unl8toiq71bak8aodwfq', 'Arn': 'arn:aws:iam::000000000000:role/teste123', 'CreateDate': datetime.datetime(2022, 8, 10, 14, 51, 18, 639000, tzinfo=tzlocal()), 'AssumeRolePolicyDocument': '{\n    "Version": "2012-10-17",\n    "Statement": [\n        {\n            "Effect": "Allow",\n            "Principal": {\n                "Service": [\n                    "ec2.amazonaws.com"\n                ]\n            },\n            "Action": [\n                "sts:AssumeRole"\n            ]\n        }\n    ]\n}', 'MaxSessionDuration': 3600}}, headers={'Content-Type': 'text/plain; charset=utf-8', 'server': 'amazon.com', 'Content-Length': '911', 'Connection': 'close', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 'Access-Control-Allow-Headers': 'authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request', 'Access-Control-Expose-Headers': 'etag,x-amz-version-id'})

2022-08-10T14:51:55.522  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS iam.UpdateAssumeRolePolicy => 400 (MalformedPolicyDocument); UpdateAssumeRolePolicyRequest({'RoleName': 'teste123', 'PolicyDocument': '{\n    "Version": "2012-10-17",\n    "Statement": [\n        {\n            "Effect": "Allow",\n            "Principal": {\n                "Service": [\n                    "ec2.amazonaws.com"\n                ]\n            },\n            "Action": [\n                "sts:AssumeRole"\n            ]\n        }\n    ]\n}'}, headers={'Host': 'localhost:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/1.22.100 Python/3.8.10 Linux/5.10.102.1-microsoft-standard-WSL2 botocore/1.25.2', 'X-Amz-Date': '20220810T145155Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=a/20220810/us-east-2/iam/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=4f8d94265133c79067e2d3dd4c07fee201aee8cf3c8df3fbce2c287b37240ef8', 'amz-sdk-invocation-id': 'b69468e9-2d23-43ce-a6f2-3b52032b4920', 'amz-sdk-request': 'attempt=1', 'Content-Length': '504', 'x-localstack-tgt-api': 'iam'}); MalformedPolicyDocument(Has prohibited field Resource., headers={'Content-Type': 'application/xml', 'server': 'amazon.com', 'X-Amzn-ErrorType': 'MalformedPolicyDocument', 'status': '400', 'Content-Length': '325', 'Connection': 'close', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 'Access-Control-Allow-Headers': 'authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request', 'Access-Control-Expose-Headers': 'etag,x-amz-version-id'})
```

File:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Service": [
                    "ec2.amazonaws.com"
                ]
            },
            "Action": [
                "sts:AssumeRole"
            ]
        }
    ]
}
```

### Expected Behavior

When updating the policy of role, it should produce a success response instead of a error

### How are you starting LocalStack?

With the `localstack` script

#### How are you starting localstack

`DEBUG=1 LS_LOG=trace localstack start`

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

```
awslocal iam get-role --role-name teste123
awslocal iam create-role --role-name teste123 --assume-role-policy-document file://Test-Role-Trust-Policy.json
awslocal iam update-assume-role-policy --role-name teste123 --policy-document file://Test-Role-Trust-Policy.json
```

### Environment

```markdown
- OS: Ubuntu 20.04
- LocalStack: latest (@sha256:01fbb996d0b94d5e1ecb3b3c131667c4e8f7a4b36ba0ad44787b5b517f643ed0)
```

### Anything else?

Testing against the version `1.0.3`, this behavior is not found and the command is executed with success

I used the following command to test this version:
```sh
docker run --rm --name localstack_main -v /home/rodjul/.cache/localstack/volume:/var/lib/localstack -v /var/run/docker.sock:/var/run/docker.sock -p 127.0.0.1:4566:4566 -p 127.0.0.1:4571:4571 -p 127.0.0.1:12121:12121 -p 127.0.0.1:4510-4559:4510-4559 -e DEFAULT_REGION=us-east-1 -e TEST_AWS_ACCOUNT_ID=000000000000 -e DEBUG=1 -e LS_LOG=trace -e DOCKER_HOST=unix:///var/run/docker.sock -e HOST_TMP_FOLDER=/home/rodjul/.local/lib/python3.8/site-packages/.filesystem/var/lib/localstack/tmp -e SET_TERM_HANDLER=1 --privileged localstack/localstack:1.0.3
```

Which results in the following logs:
```
2022-08-10T14:57:06.874  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS iam.GetRole => 404 (NoSuchEntity); GetRoleRequest({'RoleName': 'teste123'}, headers={'Host': 'localhost:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/1.22.100 Python/3.8.10 Linux/5.10.102.1-microsoft-standard-WSL2 botocore/1.25.2', 'X-Amz-Date': '20220810T145706Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=a/20220810/us-east-2/iam/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=d4eb9fd3752104caa903493648984243fbe466ec0978811eb4be757622968ff1', 'amz-sdk-invocation-id': '06e3cc40-572e-4c91-9584-0d219f2d44ce', 'amz-sdk-request': 'attempt=1', 'Content-Length': '51', 'x-localstack-tgt-api': 'iam'}); NoSuchEntity(Role teste123 not found, headers={'Content-Type': 'application/xml', 'server': 'amazon.com', 'X-Amzn-ErrorType': 'NoSuchEntity', 'status': '404', 'Content-Length': '307', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 'Access-Control-Allow-Headers': 'authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request', 'Access-Control-Expose-Headers': 'etag,x-amz-version-id'})

2022-08-10T14:57:07.166  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS iam.CreateRole => 200; CreateRoleRequest({'RoleName': 'teste123', 'AssumeRolePolicyDocument': '{\n    "Version": "2012-10-17",\n    "Statement": [\n        {\n            "Effect": "Allow",\n            "Principal": {\n                "Service": [\n                    "ec2.amazonaws.com"\n                ]\n            },\n            "Action": [\n                "sts:AssumeRole"\n            ]\n        }\n    ]\n}'}, headers={'Host': 'localhost:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/1.22.100 Python/3.8.10 Linux/5.10.102.1-microsoft-standard-WSL2 botocore/1.25.2', 'X-Amz-Date': '20220810T145707Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=a/20220810/us-east-2/iam/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=fa4623cd582e6896958f3f7e356b0477b469b9a59b98f3b20a12ea9195e80e9a', 'amz-sdk-invocation-id': 'dcb646db-70f0-446c-9a77-006a7b6e3493', 'amz-sdk-request': 'attempt=1', 'Content-Length': '502', 'x-localstack-tgt-api': 'iam'}); CreateRoleResponse({'Role': {'Path': '/', 'RoleName': 'teste123', 'RoleId': '18zloe1ce8xzg2fet070', 'Arn': 'arn:aws:iam::000000000000:role/teste123', 'CreateDate': datetime.datetime(2022, 8, 10, 14, 57, 7, 158000, tzinfo=tzlocal()), 'AssumeRolePolicyDocument': '{\n    "Version": "2012-10-17",\n    "Statement": [\n        {\n            "Effect": "Allow",\n            "Principal": {\n                "Service": [\n                    "ec2.amazonaws.com"\n                ]\n            },\n            "Action": [\n                "sts:AssumeRole"\n            ]\n        }\n    ]\n}', 'MaxSessionDuration': 3600}}, headers={'Content-Type': 'text/plain; charset=utf-8', 'server': 'amazon.com', 'Content-Length': '911', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 'Access-Control-Allow-Headers': 'authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request', 'Access-Control-Expose-Headers': 'etag,x-amz-version-id'})

2022-08-10T14:57:07.456  INFO --- [   asgi_gw_0] localstack.request.aws     : AWS iam.UpdateAssumeRolePolicy => 200; UpdateAssumeRolePolicyRequest({'RoleName': 'teste123', 'PolicyDocument': '{\n    "Version": "2012-10-17",\n    "Statement": [\n        {\n            "Effect": "Allow",\n            "Principal": {\n                "Service": [\n                    "ec2.amazonaws.com"\n                ]\n            },\n            "Action": [\n                "sts:AssumeRole"\n            ]\n        }\n    ]\n}'}, headers={'Host': 'localhost:4566', 'Accept-Encoding': 'identity', 'Content-Type': 'application/x-www-form-urlencoded; charset=utf-8', 'User-Agent': 'aws-cli/1.22.100 Python/3.8.10 Linux/5.10.102.1-microsoft-standard-WSL2 botocore/1.25.2', 'X-Amz-Date': '20220810T145707Z', 'Authorization': 'AWS4-HMAC-SHA256 Credential=a/20220810/us-east-2/iam/aws4_request, SignedHeaders=content-type;host;x-amz-date, Signature=6b808e0253173363f7c8fe7cf4ad7a82e4207e0a93a3118090442e948672bb05', 'amz-sdk-invocation-id': 'af437c39-c867-4752-987c-29ed8ee94d23', 'amz-sdk-request': 'attempt=1', 'Content-Length': '504', 'x-localstack-tgt-api': 'iam'}); Response({}, headers={'Content-Type': 'text/plain; charset=utf-8', 'server': 'amazon.com', 'Content-Length': '160', 'Access-Control-Allow-Origin': '*', 'Access-Control-Allow-Methods': 'HEAD,GET,PUT,POST,DELETE,OPTIONS,PATCH', 'Access-Control-Allow-Headers': 'authorization,cache-control,content-length,content-md5,content-type,etag,location,x-amz-acl,x-amz-content-sha256,x-amz-date,x-amz-request-id,x-amz-security-token,x-amz-tagging,x-amz-target,x-amz-user-agent,x-amz-version-id,x-amzn-requestid,x-localstack-target,amz-sdk-invocation-id,amz-sdk-request', 'Access-Control-Expose-Headers': 'etag,x-amz-version-id'})
```

**Comment by whummer** [comment]
Thanks for reporting @rodjul . This should be fixed via #6754 - can you please pull the `latest` Docker image and give it another try? Please let us know if the problem persists.. Thanks!

**Comment by rodjul** [comment]
Hello @whummer, the latest version worked. Thanks!
